commit 39e77e9d67511ac9bfbb00ab471fc7b0374dcdf6
Author: Eric Anholt <eric@anholt.net>
Date:   Mon Jan 25 14:32:41 2016 -0800

    drm/vc4: Fix -ERESTARTSYS error return from BO waits.
    
    This caused the wait ioctls to claim that waiting had completed when
    we actually got interrupted by a signal before it was done.  Fixes
    broken rendering throttling that produced serious lag in X window
    dragging.
    
    Signed-off-by: Eric Anholt <eric@anholt.net>

diff --git a/drivers/gpu/drm/vc4/vc4_gem.c b/drivers/gpu/drm/vc4/vc4_gem.c
index f8c003a..dc3044d 100644
--- a/drivers/gpu/drm/vc4/vc4_gem.c
+++ b/drivers/gpu/drm/vc4/vc4_gem.c
@@ -352,12 +352,10 @@ vc4_wait_for_seqno(struct drm_device *dev, uint64_t seqno, uint64_t timeout_ns,
 	finish_wait(&vc4->job_wait_queue, &wait);
 	trace_vc4_wait_for_seqno_end(dev, seqno);
 
-	if (ret && ret != -ERESTARTSYS) {
+	if (ret && ret != -ERESTARTSYS)
 		DRM_ERROR("timeout waiting for render thread idle\n");
-		return ret;
-	}
 
-	return 0;
+	return ret;
 }
 
 static void
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1197_39e77e9d67511ac9bfbb00ab471fc7b0374dcdf6.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1197_39e77e9d67511ac9bfbb00ab471fc7b0374dcdf6.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
